src/SDK/MonitoringApplianceEmulator/Controls/HypertextBlock.cs

```diff
@@ -0,0 +1,135 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="HypertextBlock.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Controls
+{
+    using System;
+    using System.Diagnostics;
+    using System.Text.RegularExpressions;
+    using System.Windows;
+    using System.Windows.Controls;
+    using System.Windows.Documents;
+
+    /// <summary>
+    /// An extension of <see cref="TextBlock"/> control that displays hypertext. The hypertext should be transfered using the <see cref="Hypertext"/> dependency property.
+    /// </summary>
+    public class HypertextBlock : TextBlock
+    {
+        /// <summary>
+        /// The hypertext dependency property. This text may include multiple hyperlinks in valid HTML A tags.
+        /// For example, '<a href="https://msdn.microsoft.com">Developer Network</a>' will be replaced with a hyperlink with display text of "Developers Network".
+        /// </summary>
+        public static readonly DependencyProperty HypertextProperty = DependencyProperty.Register(
```

Comment:
>HypertextProperty [](start = 50, length = 17)

is this really the best way to do it? what if someone puts in both Hypertext and Text?

Reply:
you are right. in such case both will be added to the context, but only hypertext support hyperlinks... I will add a task to improve it but let's go with this as a beginning in order not to waste time
(task 1391319)

---
In reply to: [234515826](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/25#discussion_r234515826) [](ancestors = 234515826)